src/GovUk.Education.ExploreEducationStatistics.Data.Api/Services/PermalinkService.cs

```diff
@@ -198,6 +200,43 @@ public class PermalinkService : IPermalinkService
                 );
         }
 
+        // TODO EES-3755 Remove after Permalink snapshot work is complete
+        public async Task<Either<ActionResult, Unit>> MigratePermalink(Guid permalinkId,
+            CancellationToken cancellationToken = default)
+        {
+            return await _contentDbContext.Permalinks
+                .SingleOrNotFoundAsync(
+                    predicate: permalink => permalink.Id == permalinkId && permalink.Legacy,
+                    cancellationToken: cancellationToken)
+                .OnSuccess(async permalink =>
+                {
+                    // Return a bad request result if the Permalink already has a snapshot
+                    if (permalink.LegacyHasSnapshot == true)
```

Comment:
Thought I'd caught you doing the old unnecessary `== true` there, until I realised `LegacyHasSnapshot` is nullable :)